File: hw/CEDENOR/U3/HW29Composite&TemplateMethod/Java/BeverageApp/src/main/java/ec/edu/espe/beverageapp/controller/Coffee.java
package ec.edu.espe.beverageapp.controller;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 *
 * @author Nahomi Cedeño, codecreators, DCCO - ESPE
 */
public class Coffee extends CoffeeinBeverage{


    @Override
    void brew() {
        System.out.println("Dripping coffee through filter");
    }

    @Override
    void addCondiments() {
        System.out.println("Adding sugar and milk");
    }

    @Override
    boolean wantsCondiments() {
        String answer = getUserInput();
        if (answer.toLowerCase().startsWith("y")){
            return true;
        }else{
            return false;
        }
    }
    
    private String getUserInput() {
        String answer = null;
        System.out.println("Would you like milk and sugar with your coffee (y/n)?");

        try {
            BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
            answer = in.readLine();  
        } catch (IOException e) {
            System.out.println("An error occurred while trying to read your input. Please try again.");
            e.printStackTrace(); 
        }

        return answer;
    }   
  
}

File: hw/CEDENOR/U3/HW29Composite&TemplateMethod/Python/Template/controller/Coffee.py

import sys
import os
# Nahomi Cedeño, codecreators, DCCO - ESPE

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from controller.CoffeeinBeverage import CoffeeinBeverage

class Coffee(CoffeeinBeverage):
    def brew(self):
        print("Dripping coffee through filter")

    def add_condiments(self):
        print("Adding sugar and milk")

    def wants_condiments(self):
        answer = self.get_user_input("Would you like milk and sugar with your coffee (y/n)? ")
        return answer.lower().startswith('y')

    def get_user_input(self, prompt):
        try:
            return input(prompt)
        except Exception as e:
            print("An error occurred while trying to read your input. Please try again.")
            return "no"
